monai/apps/detection/utils/anchor_utils.py

```diff
@@ -0,0 +1,409 @@
+
+    Module that generates anchors for a set of feature maps and
+    image sizes.
+
+    The module support computing anchors at multiple sizes and aspect ratios
+    per feature map.
+
+    sizes and aspect_ratios should have the same number of elements, and it should
+    correspond to the number of feature maps.
+
+    sizes[i] and aspect_ratios[i] can have an arbitrary number of elements.
+    For 2D images, anchor width and height w:h = 1:aspect_ratios[i,j]
+    For 3D images, anchor width, height, and depth w:h:d = 1:aspect_ratios[i,j,0]:aspect_ratios[i,j,1]
+
+    AnchorGenerator will output a set of sizes[i] * aspect_ratios[i] anchors
+    per spatial location for feature map i.
+
+    Args:
+        sizes: base size of each anchor.
+            len(sizes) is the number of feature maps, i.e., the number of output levels for
+            the feature pyramid network (FPN).
+            Each elment of ``sizes`` is a Sequence which represents several anchor sizes for each feature map.
+        aspect_ratios: the aspect ratios of anchors. ``len(aspect_ratios) = len(sizes)``.
+            For 2D images, each element of ``aspect_ratios[i]`` is a Sequence of float.
+            For 3D images, each element of ``aspect_ratios[i]`` is a Sequence of 2 value Sequence.
+        indexing: choose from {'xy', 'ij'}, optional
+            Cartesian ('xy') or matrix ('ij', default) indexing of output.
+            Cartesian ('xy') indexing swaps axis 0 and 1, which is the setting inside torchvision.
+            matrix ('ij', default) indexing keeps the original axis not changed.
+            See also indexing in https://pytorch.org/docs/stable/generated/torch.meshgrid.html
+
+    Reference:.
+        https://github.com/pytorch/vision/blob/release/0.12/torchvision/models/detection/anchor_utils.py
+
+    Example:
+        .. code-block:: python
+
+            # 2D example inputs for a 2-level feature maps
+            sizes = ((10,12,14,16), (20,24,28,32))
+            base_aspect_ratios = (1., 0.5,  2.)
+            aspect_ratios = (base_aspect_ratios, base_aspect_ratios)
+            AnchorGenerator(sizes, aspect_ratios)
+
+            # 3D example inputs for a 2-level feature maps
+            sizes = ((10,12,14,16), (20,24,28,32))
+            base_aspect_ratios = ((1., 1.), (1., 0.5), (0.5, 1.), (2., 2.))
+            aspect_ratios = (base_aspect_ratios, base_aspect_ratios)
+            AnchorGenerator(sizes, aspect_ratios)
+    """
+
+    __annotations__ = {"cell_anchors": List[torch.Tensor]}
+
+    def __init__(
+        self,
+        sizes: Sequence[Sequence[int]] = ((20, 30, 40),),
+        aspect_ratios: Sequence = (((0.5, 1), (1, 0.5)),),
+        indexing: str = "ij",
+    ) -> None:
+        super().__init__()
+
+        if not isinstance(sizes[0], Sequence):
+            # TODO change this
+            sizes = tuple((s,) for s in sizes)
+        if not isinstance(aspect_ratios[0], Sequence):
```

Comment:
Can we use `Sequence` to check data type?
Maybe better to use: https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/misc.py#L87

Thanks.